File: app/src/main/java/ru/contesta/pronunciationtrainer/ui/introduction/SplashScreenActivity.java
package ru.contesta.pronunciationtrainer.ui.introduction;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import ru.contesta.pronunciationtrainer.R;

public class SplashScreenActivity extends AppCompatActivity {

    Button completeSplashScreen;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);

        completeSplashScreen = (Button) findViewById(R.id.complete_splash_screen);

        completeSplashScreen.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intentToSelectNativeLanguageActivity = new Intent(SplashScreenActivity.this, SelectNativeLanguageActivity.class);
                startActivity(intentToSelectNativeLanguageActivity);
                Log.d("VALUES_CHANGED", "переход произошёл");
            }
        });
    }
}

File: app/src/main/java/ru/contesta/pronunciationtrainer/ui/introduction/SplashScreenActivity.kt
package ru.contesta.pronunciationtrainer.ui.introduction

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.view.Window
import androidx.appcompat.app.AppCompatActivity

import ru.contesta.pronunciationtrainer.R
import ru.contesta.pronunciationtrainer.TestCardSlider
import ru.contesta.pronunciationtrainer.TestInkPageIndicator

class SplashScreenActivity : AppCompatActivity() {
    private fun getSplashScreenDuration() = 2000L

    override fun onCreate(savedInstanceState: Bundle?) {
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)

        scheduleSplashScreen()
    }

    private fun scheduleSplashScreen() {
        val splashScreenDuration = getSplashScreenDuration()
        Handler().postDelayed(
                {
                    val intentToSelectNativeLanguage = Intent(SplashScreenActivity@this, TestInkPageIndicator::class.java)
                    startActivity(intentToSelectNativeLanguage)
                    finish()
                },
                splashScreenDuration
        )
    }
}